### PR TITLE
Files added

### PR DESCRIPTION
**Pull request by Ayush-kumar1**
This is the frontend for the e-commerce app version 1.Pls review it

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    </Router>
+    </div>
+    
+
+     
+    
+   
+
+    // <div className="App">
+    //   <Header/>
```
Comment (asmitzz) [inline src/App.js]: remove commented code

File: src/Context/CartContext.jsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+import axios from "axios";
+
+export const CartContext=createContext();
+
+
+export function CartProvider({children}){
+   
+    const [fav,setFav]=useState([])
+    const [product_data, setProduct_data]=useState([]);
+    // const [itemsinCart, setItemsinCart]=useState([])
```
Comment (asmitzz) [inline src/Context/CartContext.jsx]: remove commented code

File: src/Context/CartContext.jsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        const temp = await axios.get(
+          "https://backend-ecommerce-12.herokuapp.com/products"
+        );
+        // console.log(temp.data.products);
+    
+        setProduct_data(temp.data.products);
+      }
+
+      useEffect(() => {
+         getData();
```
Comment (asmitzz) [inline src/Context/CartContext.jsx]: use trycatch block here

File: src/DemoProduct/DemoProduct.css [hunk 4]
```diff
@@ -0,0 +1,51 @@
+.product {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: flex-end;
+    margin: 10px;
+    padding: 20px;
+    width: 100%;
+    max-height: 400px;
```
Comment (asmitzz) [inline src/DemoProduct/DemoProduct.css]: avoid pixels

File: src/DemoProduct/DemoProduct.css [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    min-width: 100px;
+    background-color: white;
+    z-index: 1;
+}
+
+.product_rating {
+    display: flex;
+}
+
+.product>img {
```
Comment (asmitzz) [inline src/DemoProduct/DemoProduct.css]: made class of image inside .product and then apply css

File: src/DemoProduct/DemoProduct.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import './DemoProduct.css'
+import {Link} from "react-router-dom"
+// import { useStateValue } from './StateProvider'
+
+function Product( {id,title, image, price, rating} ) {
+
+    // const[{basket},dispatch]=useStateValue();
+    // console.log("This is basket", basket);
+
+    // const addToBasket=()=>{
```
Comment (asmitzz) [inline src/DemoProduct/DemoProduct.jsx]: remove commneted code

**Review by asmitzz — CHANGES_REQUESTED** [review]

**Comment by asmitzz** [comment]
fixed netlify 404 broken link issue

**Comment by asmitzz** [comment]
display the number of items present in the cart and wishlist on the home page

**Comment by asmitzz** [comment]
when I click on add to cart then put a condition if product is added in the cart then go to cart option will show there otherwise same product will add multiple times in the cart

**Comment by asmitzz** [comment]
In the cart page disable the button when the quantity of product will be 1 or more then one otherwise quantity goes into minus

**Comment by asmitzz** [comment]
add cursor:pointer property on all the buttons and give some padding

**Comment by jugalkpatel** [comment]
add deployed site link in about section.

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import './index.css';
+import App from './App';
+import {CartProvider} from "./Context/CartContext"
+
+
+
+ReactDOM.render(
+  <React.StrictMode>
+     <CartProvider>
+
```
Comment (jugalkpatel) [inline src/index.js]: remove spaces here.

File: src/SingleContent/SingleContent.jsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import "./SingleContent.css"
+// import "https://ui-ninja.netlify.app/style.css"
+
+const SingleContent = ({id,image,name,price,wishlist,quantity,stock,delivery,Complete_item}) => {
+
+   const {setFav, dispatch,}=useCart();
+    return (
+        <div className="card">
+             
+             <img className="poster" src={image} alt=""/>
```
Comment (jugalkpatel) [inline src/SingleContent/SingleContent.jsx]: add text in the alt attribute.

File: src/Pages/Home/Home.jsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+import DemoProduct from '../../DemoProduct/DemoProduct'
+import {Link} from "react-router-dom";
+
+function Home() {
+    return (
+        <div className="home">
+            <div className="home_container">
+
+               <Link to="/products">
+            <img className="home_img" src="https://images-eu.ssl-images-amazon.com/images/G/31/img21/Fashion/Gateway/SuperSale_Herotator_1st-3rdMar/Rev_FST_GW_PC_BUNK_1500x600._CB655793405_.jpg" alt="" srcset=""/>
```
Comment (joyguptaa) [inline src/Pages/Home/Home.jsx]: The image is not responsive.

File: src/Header/Header.jsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import "./Header.css"
```
Comment (jugalkpatel) [inline src/Header/Header.jsx]: header is overlapping landing page content.

File: src/Header/Header.jsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+import {Link} from "react-router-dom"
+
+
+const Header = () => {
+    return (
+        <div className="header">
+            <div className="logo_position"> 
+
+            <Link to="/">
+            <img src={logo1} className="logo" alt="" />
```
Comment (jugalkpatel) [inline src/Header/Header.jsx]: add text to alt attribute.

File: src/Header/Header.jsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+           </Link>
+
+           <Link to="/cart">
+           <ShoppingCartIcon color="action" fontSize="large"/>
+           
+           </Link>
+           </div>
+           
+
+           {/* <div>
```
Comment (jugalkpatel) [inline src/Header/Header.jsx]: remove unnecessary lines.

File: src/DemoProduct/DemoProduct.css [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    max-width: 200px;
+    width: 100%;
+    object-fit: contain;
+    margin-bottom: 15px;
+}
+
+.product>button {
+    background: #f0c14b;
```
Comment (jugalkpatel) [inline src/DemoProduct/DemoProduct.css]: add color to variables.

File: src/DemoProduct/DemoProduct.css [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    border: 1px solid black;
+    margin-top: 10px;
+    color: black;
+}
+
+.product_button {
+    background: #f0c14b;
```
Comment (jugalkpatel) [inline src/DemoProduct/DemoProduct.css]: add color to variables.

File: src/Pages/Cart/Cart.jsx [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+                <div className="card_for_cart">
+
+                    <h2>{elem.name}</h2>
+                    <img className="cart_poster" src={elem.image} alt="" />
+                    <p>Rs. {elem.price}</p>
+                    <p>Quantity- {elem.quantity+ 1}</p>
+
+                    <button onClick={()=> dispatch({type: "INCREMENT", payload: elem._id})}>+</button>
+
+                    <button onClick={()=> dispatch({type: "DECREMENT", payload: elem._id})}> -- </button>
```
Comment (joyguptaa) [inline src/Pages/Cart/Cart.jsx]: It should remove the item from the cart when its count goes below 1.

File: src/DemoProduct/DemoProduct.jsx [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+
+                <div className="product_rating">
+                    {Array(rating).fill().map((_,i)=>(
+                      <p>⭐</p>
+                    ))}
+                 
+                </div>
+            </div>
+
+            <img className="demo_image"  src={image} alt=""/>
```
Comment (jugalkpatel) [inline src/DemoProduct/DemoProduct.jsx]: add text to alt attribute.

**Comment by joyguptaa** [comment]
- Wishlist page is getting broken. Please check: Add to wishlist -> Go to wishlist page -> Click on Remove from Wishlist.
Now there won't be any header and if you refresh the page it gets broken.

**Comment by jugalkpatel** [comment]
app became blank while removing item from wishlist page.
see this
https://shareit.video/TEFYfdieOSryLCyBBYf3

**Comment by jugalkpatel** [comment]
quantity and amount are going into minus.
image

**Comment by jugalkpatel** [comment]
- make code formatting better.
- remove unnecessary spaces.
- add count on cart and wishlist icons.
- fix cart duplication and add remove from cart button

Suggestions:

- add clear all filters button.
- show total products count on product listing page.

File: src/Components/Login/Login.jsx [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+            color="secondary"
+            style={{ width: "2rem", margin: "0px auto" }}
+            onClick={() => postData()}
+          >
+            Login
+          </Button>
+
+          <div className="signup-redirect">
+            <Link to="/signup">
+              {" "}
```
Comment (thesudeshdas) [inline src/Components/Login/Login.jsx:89]: Why the space?

File: src/Components/Signup/Signup.jsx [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+            onChange={(e) => setName(e.target.value)}
+          />
+          <Input
+            type="text"
+            placeholder="E-mail"
+            value={email}
+            onChange={(e) => setEmail(e.target.value)}
+          />
+          <Input
+            type="text"
```
Comment (thesudeshdas) [inline src/Components/Signup/Signup.jsx:92]: The type should be password here

File: src/Components/Signup/Signup.jsx [hunk 19]
```diff
@@ -0,0 +1,135 @@
+import React, { useState,useEffect } from "react";
```
Comment (thesudeshdas) [inline src/Components/Signup/Signup.jsx:1]: When I signed up, the async call was made, but by that time, I clkicked in sign up 2 more times. Make a loader to let user know that he is being signed up, and he should wait.
Screenshot from 2021-07-07 19-48-48

File: src/Components/Header/Header.jsx [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+    const navigate=useNavigate()
+
+function Logout(){
+    localStorage.clear();
+    dispatch({type:"CLEAR"})
+    navigate("/login")
+}
+     
+    return (
+        <div className="nav">
```
Comment (thesudeshdas) [inline src/Components/Header/Header.jsx:23]: The whole navbar has cursor pointer. It should only be on the links

File: src/Components/Header/Header.jsx [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+            <Link to={state?"/home":"/login"}>
+            <h2 className="brand-logo" >SnatchKart</h2>
+            </Link>
+             
+             <div style={{display:"flex",gap:"1rem",paddingTop:"1rem"}}>
```
Comment (thesudeshdas) [inline src/Components/Header/Header.jsx:28]: This div is not aligned vertically properly, align it to center?

File: src/Components/Wishlist/Wishlist.jsx [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+  };
+
+  return (
+    <div>
+      <div className="left-body">
+        <h1>Wishlist</h1>
+
+        {
+        data.length===0
+        ? <h2>You cart is empty</h2>
```
Comment (thesudeshdas) [inline src/Components/Wishlist/Wishlist.jsx:48]: It should be "Your wishlist is empty"

File: src/Components/Header/Header.jsx [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+                     <FavoriteIcon/>
+                     </Link>
+                 </IconButton>
+                 <IconButton>
+                     <Link to={state?"/cart":"/login"}>
+                     <ShoppingCartIcon />
+                     </Link>
+                 </IconButton>
+                 <IconButton>
+                     <Link to="/login">
```
Comment (thesudeshdas) [inline src/Components/Header/Header.jsx:50]: If I am already logged in, Why should I be redirected to the login page again? Shouldn't this either give the account details or give me an option to log out?

File: src/Components/Home/Home.jsx [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+            </div>
+
+            <div className="body_right">
+                 <img className="home_img" src={shopping} alt="" />
+            </div>
+
+            
+        </div>
+        <Link to="/product">
+        <button className="product_button">BROWSE PRODUCT</button>
```
Comment (thesudeshdas) [inline src/Components/Home/Home.jsx:35]: Give this button a cursor: pointer

File: src/App.js [hunk 25]
```diff
@@ -0,0 +1,49 @@
+import "./App.css";
```
Comment (thesudeshdas) [inline src/App.js:1]: The app breaks on reload

File: src/Components/Product/Product.jsx [hunk 26]
```diff
@@ -0,0 +1,261 @@
...
+          onChange={()=> dispatch({type:"SORT", payload:"LOW_TO_HIGH"})} />
+          Low-to-high
+          </label>
+          </div>
+
+          <hr />
+
+          <div className="filter-body">
+            
+            <h2>Ratings</h2>
```
Comment (thesudeshdas) [inline src/Components/Product/Product.jsx:177]: products list not changing on the selected rating

File: src/Components/Product/Product.jsx [hunk 27]
```diff
@@ -0,0 +1,261 @@
...
+
+            <label>
+              <input 
+              type="checkbox"
+              onChange={()=>dispatch({type:"Filter-instock"})} />
+              exclude out of stock
+            </label>
+ 
+ 
+             <label>
```
Comment (thesudeshdas) [inline src/Components/Product/Product.jsx:209]: Both the options should be checkable simultaneously

File: src/Components/Product/Product.jsx [hunk 28]
```diff
@@ -0,0 +1,261 @@
...
+       })
+       .then(res=>res.json())
+       .then(result=>{
+         console.log(result);
+       })
+  }
+
+  return (
+    <div>
+      <div className="sidenav">
```
Comment (thesudeshdas) [inline src/Components/Product/Product.jsx:152]: Create sidenav as a separate component

File: src/Components/Product/Product.jsx [hunk 29]
```diff
@@ -0,0 +1,261 @@
...
+
+          </div>
+          
+        </div>
+      </div>
+
+      <div className="body">
+        {newarr &&
+          newarr.map((elem) => {
+            return (
```
Comment (thesudeshdas) [inline src/Components/Product/Product.jsx:224]: Make card cursor: pointer

File: src/Components/Product/Product.jsx [hunk 30]
```diff
@@ -0,0 +1,261 @@
...
+              <div className="media_card_product">
+               
+                <img
```
Comment (thesudeshdas) [inline src/Components/Product/Product.jsx:227]: Give image height, width. [Object-fit: cover](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit)

File: src/Components/Product/Product.jsx [hunk 31]
```diff
@@ -0,0 +1,261 @@
...
+
+                <div
+                  style={{
+                    display: "flex",
+                    gap: "0.5rem",
+                    paddingLeft: "6.5rem",
+                  }}
+                >
+                  <IconButton onClick={()=>addtowishlist(elem.name,elem.rating,elem.price,elem.image)}>
+                    <FavoriteIcon color="inherit" />
```
Comment (thesudeshdas) [inline src/Components/Product/Product.jsx:245]: If product is successfully added to the wishlist, change the color of the icon

File: src/Components/Cart/Cart.jsx [hunk 32]
```diff
@@ -0,0 +1,215 @@
...
+
+                      <div
+                        style={{
+                          display: "flex",
+                          gap: "0.5rem",
+                          paddingTop: "2rem",
+                        }}
+                      >
+                        <IconButton>
+                          <AddIcon onClick={() => incrementer(elem._id)} />
```
Comment (thesudeshdas) [inline src/Components/Cart/Cart.jsx:164]: plus icon on the right, minus on the left

File: public/index.html [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (thesudeshdas) [inline public/index.html:27]: Change the title with change in page url & use favicon

**Review by thesudeshdas — CHANGES_REQUESTED** [review]
Look into your folder structure

**Pull request merged**